---
title: "Figure 4"
author: "Andrew Dominguez"
date: "7/23/2020"
output: html_document
---

```{r, setup, include=FALSE}
knitr::opts_knit$set(root.dir = '~/Downloads/Lovegrass_root_zone_microbiome/')
```

```{r}
library(ape)
library(vegan)
library(plyr)
library(dplyr)
library(tidyr)
library(scales)
library(grid)
library(reshape2)
library(phyloseq)
library(magrittr)
library(ggplot2)
library(ggpubr)
library(data.table)
library(labdsv)
otu = read.table(file="../Data/LGasv16SUN3.otu_table.txt", header=T, sep='\t',row.names = 1)
head(otu)
dim(otu)
tax <- read.table(file="../Data/taxonomy.fix.16S.tsv", sep='\t', header=TRUE,row.names = 1)
head(tax)
dim(tax)
tax <- select(tax,-Confidence)
tax_filtered = separate(tax, Taxon, c("Kingdom","Phylum","Class","Order", "Family", "Genus","Species"), sep= ";", remove=TRUE)
head(tax_filtered)
taxmat <- as(as.matrix(tax_filtered),"matrix")
TAX = tax_table(taxmat)
otumat <- as(as.matrix(otu), "matrix")
mode(otumat) <- "numeric"
OTU = otu_table(otumat, taxa_are_rows = TRUE)
otumat
meta0 = read.table("../Data/Andrew16S_mappingfile.txt",
                  header=TRUE,row.names=1,
                  sep="\t",stringsAsFactors=FALSE)
head(meta0)
meta <- meta0[which(meta0$Owner %in% c("Andrew")),]
head(meta)
dim(meta)
sampleData <- sample_data(meta)
physeq = phyloseq(OTU,TAX,sampleData)
physeq
physeq.prune = prune_taxa(taxa_sums(physeq) > 0, physeq)
physeq
readcount = data.table(as(sample_data(physeq.prune), "data.frame"),
                 TotalReads = sample_sums(physeq.prune), 
                 keep.rownames = TRUE)
setnames(readcount, "rn", "SampleID")
readcount = readcount[order(readcount$TotalReads), c("SampleID", "TotalReads")]
head(readcount)
set.seed(711)
physeq.prune.rarefy = rarefy_even_depth(physeq.prune, sample.size = 37702, replace = FALSE, trimOTUs = TRUE)
physeq.prune.rarefy
bacteria <- subset_taxa(physeq.prune.rarefy, Kingdom == "Bacteria")
bacteria.con <- subset_samples(bacteria, Treatment == "Control")
bacteria.june <- subset_samples(bacteria, Date == "June_2017")
bacteria.nov <- subset_samples(bacteria, Date == "November_2017")
bacteria.june.control <- subset_samples(bacteria.june, Treatment == "Control")
bacteria.nov.control <- subset_samples(bacteria.nov, Treatment == "Control")
bacteria.june.erle <- subset_samples(bacteria.june, Grass == "ERLE")
bacteria.nov.erle <- subset_samples(bacteria.nov, Grass == "ERLE")
bacteria.june.boer <- subset_samples(bacteria.june, Grass == "BOER")
bacteria.nov.boer <- subset_samples(bacteria.nov, Grass == "BOER")
```

#Bacteria

```{r}
boxplot.bacteria <- plot_richness(bacteria.nov, x = "Treatment", color = "Grass", measures = c("Observed","Chao1")) + 
  geom_boxplot() + theme_bw(base_size = 20) + theme(axis.text.x = element_text(angle = 90))
```

```{r}
boxplot.bacteria
```

##Bacterial stats

```{r}
alpha.bacteria.nov <- estimate_richness(bacteria.nov, measures = c("Observed", "Chao1","Simpson"))
alpha.bacteria.nov2 <- cbind(sample_data(bacteria.nov), alpha.bacteria.nov)
mod.10 <- aov(Observed ~ Treatment * Grass, alpha.bacteria.nov2)
summary(mod.10)
mod.11 <- aov(Chao1 ~ Treatment * Grass, alpha.bacteria.nov2)
summary(mod.11)
mod.12 <- aov(Simpson ~ Treatment * Grass, alpha.bacteria.nov2)
summary(mod.12)
```

```{r}
TukeyHSD(mod.10)
```

```{r}
TukeyHSD(mod.11)
```


```{r}
head(alpha.bacteria.nov2)
```

```{r}
alpha.bacteria.nov2$Treatment <- as.factor(alpha.bacteria.nov2$Treatment)
alpha.bacteria.nov2$Grass <- as.factor(alpha.bacteria.nov2$Grass)
```

```{r}
alpha.bacteria.nov2.erle <- subset(alpha.bacteria.nov2,Grass == "ERLE")
```

```{r}
kw.1 <- kruskal.test(Observed ~ Treatment,data = alpha.bacteria.nov2.erle)
kw.1
kw.2 <-kruskal.test(Chao1 ~ Treatment,data = alpha.bacteria.nov2.erle)
kw.2
```

```{r}
alpha.bacteria.nov2.boer <- subset(alpha.bacteria.nov2,Grass == "BOER")
```

```{r}
kw.1 <- kruskal.test(Observed ~ Treatment,data = alpha.bacteria.nov2.boer)
kw.1
kw.2 <-kruskal.test(Chao1 ~ Treatment,data = alpha.bacteria.nov2.boer)
kw.2
```

```{r}
otu = read.table(file="../Data/FungiAD.otu_table.txt", header=T, sep='\t',row.names = 1)
head(otu)
dim(otu)

tax <- read.table(file="../Data/FungiAD.ASVs.taxonomy.fix.txt", sep='\t', header=TRUE,row.names = 1)
head(tax)

dim(tax)

taxmat <- as(tax,"matrix")
head(taxmat)

TAX = tax_table(taxmat)

otumat <- as(as.matrix(otu), "matrix")
mode(otumat) <- "numeric"
OTU = otu_table(otumat, taxa_are_rows = TRUE)

otumat

meta0 = read.table("../Data/Andrew_ITS.mapping_file.txt",
                  header=TRUE,row.names=1,
                  sep="\t",stringsAsFactors=FALSE)
meta0

sampleData <- sample_data(meta0)


physeq = phyloseq(OTU,TAX,sampleData)



physeq


physeq.prune = prune_taxa(taxa_sums(physeq) > 0, physeq)
physeq

readcount = data.table(as(sample_data(physeq.prune), "data.frame"),
                 TotalReads = sample_sums(physeq.prune), 
                 keep.rownames = TRUE)
setnames(readcount, "rn", "SampleID")


readcount = readcount[order(readcount$TotalReads), c("SampleID", "TotalReads")]
head(readcount)

set.seed(711)
physeq.prune.rarefy = rarefy_even_depth(physeq.prune, sample.size = 4941, replace = FALSE, trimOTUs = TRUE)
physeq.prune.rarefy

physeq.prune.rarefy.fun <- subset_taxa(physeq.prune.rarefy,Kingdom == "Fungi")
physeq.prune.rarefy.fun

fun.con <- subset_samples(physeq.prune.rarefy.fun, Treatment == "Control")
fun.june <- subset_samples(physeq.prune.rarefy.fun, Date == "June_2017")
fun.nov <- subset_samples(physeq.prune.rarefy.fun, Date == "November_2017")
fun.june.control <- subset_samples(fun.june, Treatment == "Control")
fun.nov.control <- subset_samples(fun.nov, Treatment == "Control")
fun.june.erle <- subset_samples(fun.june, Grass == "ERLE")
fun.june.boer <- subset_samples(fun.june, Grass == "BOER")
fun.nov.erle <- subset_samples(fun.nov, Grass == "ERLE")
fun.nov.boer <- subset_samples(fun.nov, Grass == "BOER")
```

#Fungi

```{r}
otu = read.table(file="../Data/FungiAD.otu_table.txt", header=T, sep='\t',row.names = 1)
head(otu)
dim(otu)

tax <- read.table(file="../Data/FungiAD.ASVs.taxonomy.fix.txt", sep='\t', header=TRUE,row.names = 1)
head(tax)

dim(tax)

taxmat <- as(tax,"matrix")
head(taxmat)

TAX = tax_table(taxmat)

otumat <- as(as.matrix(otu), "matrix")
mode(otumat) <- "numeric"
OTU = otu_table(otumat, taxa_are_rows = TRUE)

otumat

meta0 = read.table("../Data/Andrew_ITS.mapping_file.txt",
                  header=TRUE,row.names=1,
                  sep="\t",stringsAsFactors=FALSE)
meta0

sampleData <- sample_data(meta0)


physeq = phyloseq(OTU,TAX,sampleData)



physeq


physeq.prune = prune_taxa(taxa_sums(physeq) > 0, physeq)
physeq

readcount = data.table(as(sample_data(physeq.prune), "data.frame"),
                 TotalReads = sample_sums(physeq.prune), 
                 keep.rownames = TRUE)
setnames(readcount, "rn", "SampleID")


readcount = readcount[order(readcount$TotalReads), c("SampleID", "TotalReads")]
head(readcount)

set.seed(711)
physeq.prune.rarefy = rarefy_even_depth(physeq.prune, sample.size = 4941, replace = FALSE, trimOTUs = TRUE)
physeq.prune.rarefy

physeq.prune.rarefy.fun <- subset_taxa(physeq.prune.rarefy,Kingdom == "Fungi")
physeq.prune.rarefy.fun

fun.con <- subset_samples(physeq.prune.rarefy.fun, Treatment == "Control")
fun.june <- subset_samples(physeq.prune.rarefy.fun, Date == "June_2017")
fun.nov <- subset_samples(physeq.prune.rarefy.fun, Date == "November_2017")
fun.june.control <- subset_samples(fun.june, Treatment == "Control")
fun.nov.control <- subset_samples(fun.nov, Treatment == "Control")
fun.june.erle <- subset_samples(fun.june, Grass == "ERLE")
fun.june.boer <- subset_samples(fun.june, Grass == "BOER")
fun.nov.erle <- subset_samples(fun.nov, Grass == "ERLE")
fun.nov.boer <- subset_samples(fun.nov, Grass == "BOER")
```

```{r}
boxplot.fungi <- plot_richness(fun.nov, x = "Treatment", color = "Grass", measures = c("Observed","Chao1")) +
  geom_boxplot() + theme_bw(base_size = 20) + theme(axis.text.x = element_text(angle = 90))
```

```{r}
boxplot.fungi
```

```{r}
figure4 <- ggarrange(boxplot.bacteria, boxplot.fungi,
                    labels = c("A", "B"),
                    ncol = 1, nrow = 2)
```



```{r}
figure4
```

```{r}
figure4
ggsave("Figure4.png",width=12,height=10,dpi = 400)
```

```{r warning=FALSE}
pdf("../Figures/Figure_4.pdf", width = 12, height = 10 )
figure4
dev.off()
```

##Fungal stats

```{r}
alpha.fun.nov <- estimate_richness(fun.nov, measures = c("Observed", "Chao1","Simpson"))
alpha.fun.nov2 <- cbind(sample_data(fun.nov), alpha.fun.nov)
mod.10 <- aov(Observed ~ Treatment * Grass, alpha.fun.nov2)
summary(mod.10)
mod.11 <- aov(Chao1 ~ Treatment * Grass, alpha.fun.nov2)
summary(mod.11)
mod.12 <- aov(Simpson ~ Treatment * Grass, alpha.fun.nov2)
summary(mod.12)
```

```{r}
TukeyHSD(mod.10)
```

```{r}
TukeyHSD(mod.11)
```


```{r}
head(alpha.fun.nov2)
```

```{r}
alpha.fun.nov2$Treatment <- as.factor(alpha.fun.nov2$Treatment)
alpha.fun.nov2$Grass <- as.factor(alpha.fun.nov2$Grass)
```

```{r}
alpha.fun.nov2.erle <- subset(alpha.fun.nov2,Grass == "ERLE")
```

```{r}
kw.1 <- kruskal.test(Observed ~ Treatment,data = alpha.fun.nov2.erle)
kw.1
kw.2 <-kruskal.test(Chao1 ~ Treatment,data = alpha.fun.nov2.erle)
kw.2
```

```{r}
alpha.fun.nov2.boer <- subset(alpha.fun.nov2,Grass == "BOER")
```

```{r}
kw.1 <- kruskal.test(Observed ~ Treatment,data = alpha.fun.nov2.boer)
kw.1
kw.2 <-kruskal.test(Chao1 ~ Treatment,data = alpha.fun.nov2.boer)
kw.2
```